src/d3a/models/strategy/settlement/strategy.py

```diff
@@ -125,31 +125,40 @@ def event_market_cycle(self, strategy: BaseStrategy) -> None:
                     )
                 except MarketException:
                     pass
+        # has to be called in order not to update the initial bid/offer in the first tick
+        self.bid_updater.increment_update_counter_all_markets(strategy)
+        self.offer_updater.increment_update_counter_all_markets(strategy)
 
-    def event_tick(self, strategy: BaseStrategy) -> None:
+    def event_tick(self, strategy: BidEnabledStrategy) -> None:
         """
-        Update posted settlement bids and offers on market tick
+        Update posted settlement bids and offers on market tick.
+        Order matters here:
+            - FIRST: the bis and offers need to be updated (update())
+            - SECOND: the update counter has to be increased (increment_update_counter_all_markets)
         Args:
             strategy: Strategy object of the asset
 
         Returns: None
 
         """
-        self.bid_updater.increment_update_counter_all_markets(strategy)
-        self.offer_updater.increment_update_counter_all_markets(strategy)
         for market in strategy.area.settlement_markets.values():
             self.bid_updater.update(market, strategy)
             self.offer_updater.update(market, strategy)
 
-    def _get_settlement_market_by_id(
-            self, strategy: BaseStrategy, market_id: str) -> Optional["Market"]:
+        self.bid_updater.increment_update_counter_all_markets(strategy)
+        self.offer_updater.increment_update_counter_all_markets(strategy)
+
+    @staticmethod
+    def _get_settlement_market_by_id(strategy: BidEnabledStrategy,
+                                     market_id: str) -> Optional["Market"]:
         markets = [m for m in strategy.area.settlement_markets.values() if m.id == market_id]
```

Comment:
Nit
```suggestion
        markets = [market_slot for market_slot in strategy.area.settlement_markets.values() if market_slot.id == market_id]
```
I know this is more verbose, but we discussed about the need to refactor our code to reduce ambiguity between markets, market slots, etc.

Reply:
I link to this discussion here https://github.com/gridsingularity/d3a-interface/pull/246#discussion_r705945794